### PR TITLE
Add CloudStack support

### PR DESCRIPTION
**Pull request by romain-dartigues**
This is the third and final PR to add [CloudStack](https://cloudstack.apache.org/) support into BOSH.

The two previous were:

- https://github.com/cloudfoundry/bosh-deployment/pull/424
- https://github.com/cloudfoundry/jumpbox-deployment/pull/33

Please let me know how to improve this MR.

### CODE REVIEW

**Comment by ramonskie** [comment]
i noticed that you are relying on packr. and we migrated away from packr to go embed 
see https://github.com/cloudfoundry/bosh-bootloader/pull/549

**Comment by romain-dartigues** [comment]
Thank you for your review @ramonskie!
Indeed, the code was done before you moved out packr, it has been corrected.

File: go.mod [hunk 1]
```diff
@@ -22,16 +22,17 @@ require (
...
-	google.golang.org/api v0.1.0
+	golang.org/x/oauth2 v0.0.0-20210402161424-2e8d93401602
+	google.golang.org/api v0.44.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
 require (
-	cloud.google.com/go v0.36.0 // indirect
+	cloud.google.com/go v0.81.0 // indirect
+	cloud.google.com/go/storage v1.10.0 // indirect
```
Comment (ramonskie) [inline go.mod:35]: this could potentially need some more testing

**Review by ramonskie — APPROVED** [review]

**Comment by jpalermo** [comment]
Thanks! A new release should be cut if everything passes.

**Comment by romain-dartigues** [comment]
@ramonskie < Sorry, I didn't see you  suggestions in time.

About the comment on go.mod; it is not the right place, but if you have the energy to enlighten me about it please?
I'm just doing a `go mod tidy && go mod vendor`; should I hand-fix the go.mod afterwards?

> `if name != "runtime-config.yml" && strings.HasSuffix(name, ".yml") {`
> could this be in a separate pr?

Indeed, it should have been separated…
It was a leftover of the rework of our patches in order to give it back to the community; my apologies to have missed it.

**Comment by ramonskie** [comment]
this pr with bumping the google`google.golang.org/api` has broken our unit tests.
 it seems that there is some refactoring required
 ```
NewClient [It] works
   [FAILED] Unexpected error:
      <*errors.errorString | 0xc0004d0060>: 
      get region: googleapi: got HTTP response code 404 with body: 
      {
          s: "get region: googleapi: got HTTP response code 404 with body: ",
      }
  occurred
  In [It] at: bosh-bootloader/gcp/client_provider_test.go:65 @ 12/29/23 11:32:13.531
```
i also noticed that some functions like `compute.New` are now deprecated

**Comment by ramonskie** [comment]
i reverted the go mod vendor commit see 
https://github.com/cloudfoundry/bosh-bootloader/commit/493a1a9423193d76db93cdac2026dfff8b772218

this temporary solves our issue so we can do release of the bootloader. but it probably breaks cloudstack as i have no possibility to test it locally

**Comment by romain-dartigues** [comment]
@ramonskie < thank you for your indefectible support, we'll fix it before our next increment (I'll poke this discussion when it's done).

**Pull request merged**